### ISSUE TITLE
Failed on creating cephobjectstoreusers 

### ISSUE THREAD

**Issue by yanchicago** [rook/rook#7353]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
Failed on creating cephobjectstoreusers. After apply "object-user.yaml", there's cephobjectstoreusers created but its secret is not created. And within ceph, the user is not seen by "radosgw-admin user list". 
```
[root@us-401-control-01 /var/log]# knc get cephobjectstore rook-ceph-store
NAME              AGE
rook-ceph-store   174d
[root@us-401-control-01 /var/log]# knc get cephobjectstoreusers my-user 
NAME      AGE
my-user   41m
[root@us-401-control-01 /var/log]# knc get secret rook-ceph-object-user-rook-ceph-store-my-user
Error from server (NotFound): secrets "rook-ceph-object-user-rook-ceph-store-my-user" not found
```

**Expected behavior:**
The user and its secret are created together. 

**How to reproduce it (minimal and precise):**

The issue may be related to the operator in "op-osd: waiting on orchestration status update from 1 remaining nodes" state. And the cephcluster is in "Creating" state with "HEALTH_OK" state. 

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
* Operator's logs, if necessary 
[op_restart.log.txt](https://github.com/rook/rook/files/6079848/op_restart.log.txt)
[operator_before_restart.txt](https://github.com/rook/rook/files/6079850/operator_before_restart.txt)

* Crashing pod(s) logs, if necessary

 To get logs, use `kubectl -n <namespace> logs <pod name>`
When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
Read [Github documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Environment**:
* OS (e.g. from /etc/os-release): CentOS-7
* Kernel (e.g. `uname -a`): Linux us-401-control-01 4.18.0-147.5.1.el8_1.x86_64 #1 SMP Wed Feb 5 02:00:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
* Cloud provider or hardware configuration: Openstack
* Rook version (use `rook version` inside of a Rook Pod): v1.2.7
* Storage backend version (e.g. for ceph do `ceph -v`): 14.2.8
* Kubernetes version (use `kubectl version`): v1.17.4
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): Tectonic
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)): HEALTH_OK

**Comment by travisn** [comment]
@yanchicago It appears the operator is not reconciling the object store users. Before the restart, there are errors about the watcher failing, then after the restart, the operator is getting stuck reconciling the OSDs. Back in Rook v1.2, the operator would wait for the OSDs to complete the reconcile before starting to reconcile the other CRs such as the object users. 

So the question is why the OSDs are not completing now. Do you still see the osd prepare pods in running state? If so, the operator is waiting for them to complete. Or if you have removed any nodes from the cluster, it's possible you have a stale osd status configmap that the operator thinks it needs to wait for a node that no longer exists. In that case, we can clean up those extra configmaps, and may need to restart the operator again after that.

Newer Rook releases have improved behavior around not getting stuck like this.

**Comment by yanchicago** [comment]
@travisn Yes, you're correct. After I deleted one of prepare job, the orchestrating completed and the object user was created successfully. Many thanks.

**Comment by travisn** [comment]
> @travisn Yes, you're correct. After I deleted one of prepare job, the orchestrating completed and the object user was created successfully. Many thanks.

Ok sounds good, we can close this issue then?

**Comment by yanchicago** [comment]
Yes. Thanks.
